[MIG] mail_activity_team: Migration to 14.0

SodexisTeam opened the pull request:
Depends on https://github.com/OCA/social/pull/661

@stephankeller

JordiBForgeFlow committed:
[ADD] mail_activity_team

MiquelRForgeFlow committed:
[IMP] Some improvements and fixes

MiquelRForgeFlow committed:
[ADD] tests

MiquelRForgeFlow committed:
[MIG] mail_activity_team: Migration to 12.0

AdriaGForgeFlow committed:
[IMP] Set team responsible as default user when activity is created

AdriaGForgeFlow committed:
[IMP] Add search filter for My Team Activities

bealdav committed:
FIX mail_activ_team: always allow superuser

oca-travis committed:
[UPD] Update mail_activity_team.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

pegonzalezspesol committed:
[IMP] mail_activity_team: black, isort

pegonzalezspesol committed:
[MIG] 13.0 mail_activity_team

oca-travis committed:
[UPD] Update mail_activity_team.pot

OCA-git-bot committed:
[UPD] README.rst

MiquelRForgeFlow committed:
[UPD] Eficent -> ForgeFlow

alessandro-fiorino committed:
Added translation using Weblate (Italian)

alessandro-fiorino committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (24 of 24 strings)

Translation: social-13.0/social-13.0-mail_activity_team
Translate-URL: https://translation.odoo-community.org/projects/social-13-0/social-13-0-mail_activity_team/it/

Commit:
[IMP] pre-commit run -a

CasVissers-360ERP commented:
I think there is a missing parameter in this module, on base.automation there is no option to set an activity team if state = next_activity. This results in "The assigned user is not member of the team %s " because an incorrect team might be set on the activity.

Demo use case:
1. Create automated action with type "Next activity" for user A
2. Create a ticket with no user assigned
3. If multiple team are in the system the wrong team might be set

OCA-git-bot committed:
[UPD] README.rst

atchuthan commented:
@CasVissers-360ERP We couldn't reproduce this issue.  Could you provide us a video describing the problem such that we can resolve it?

CasVissers-360ERP commented on mail_activity_team/models/mail_activity.py:
```diff
@@ -0,0 +1,72 @@
+            and a.user_id
+            and a.user_id not in a.team_id.member_ids
+        ):
+            raise ValidationError(_("The assigned user is not member of the team."))
```
Might be better to add user to error?
`raise ValidationError(_("The assigned user %s is not member of the team %s.") % (self.user_id.name, self.team_id.name))`

atchuthan replied:
added

chienandalu committed:
[FIX] mail_activity_team: scheduled activity

When an activity is scheduled due to automatic processes the onchanges
won't adapt the proper user team, so we could have a mismatch that
provokes an exceptions due to the designed constaints

chienandalu reviewed (changes requested):
Please include this commit https://github.com/OCA/social/commit/6a96684e50a12b067d03eb7da381a5b812e1fa04

It should fix the issue commented by @CasVissers-360ERP

pedrobaeza commented:
Please include #697 and fix CI.

oca-travis committed:
[UPD] Update mail_activity_team.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
mail_activity_team 13.0.1.0.1

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: social-13.0/social-13.0-mail_activity_team
Translate-URL: https://translation.odoo-community.org/projects/social-13-0/social-13-0-mail_activity_team/

chienandalu committed:
[FIX] mail_activity_team: archived users

- If a user is archived but there's an automatic activity, the team
check wouldn't locate him. We have to ensure the context.
- When the automatic activity gets a user with no team, we should set
the team to a void one as well.

TT30470

MiquelRForgeFlow commented:
And also include https://github.com/OCA/social/pull/729.

OCA-git-bot committed:
mail_activity_team 13.0.1.0.2

atchuthan commented:
> Please include #697 and fix CI.

added the fix with commit https://github.com/OCA/social/pull/662/commits/67792ea5f6b4836408e9d0f6bd03f5d296cf3342

atchuthan commented:
> And also include #729.

added the fix with commit https://github.com/OCA/social/pull/662/commits/32e9a4145b91329626ab2cf8565b038cafb7c13d

AaronHForgeFlow commented:
Hi @atchuthan can you attend travis here? Are you still working on this? Thank you.

atchuthan commented:
@AaronHForgeFlow Updated latest code from 14.0 branch

MiquelRForgeFlow commented:
@atchuthan please write:

`/ocabot rebase`

AaronHForgeFlow commented:
What is the status of this PR at this moment?

atchuthan commented:
> What is the status of this PR at this moment?

It is working in one of our customers successfully.  We will try to complete the test file in few days.

JoanMForgeFlow reviewed (approved):
LGTM!

AaronHForgeFlow commented on mail_activity_team/tests/test_mail_activity_team.py:
```diff
@@ -139,15 +139,6 @@ def test_missing_activities(self):
         self.assertEqual(self.team1.count_missing_activities, 0)
         self.assertEqual(self.act1.team_id, self.team1)
 
-    def test_team_onchanges(self):
```
why is this removed?

atchuthan replied:
When a new member is added, the onchange clears the value in leader(i.e. set to null), so we removed it in migration.

AaronHForgeFlow reviewed (approved):
Functional review LGTM

AaronHForgeFlow commented:
@chienandalu could you update your review? thanks

tfossoul committed:
Added translation using Weblate (French)

tfossoul committed:
Translated using Weblate (French)

Currently translated at 32.0% (8 of 25 strings)

Translation: social-13.0/social-13.0-mail_activity_team
Translate-URL: https://translation.odoo-community.org/projects/social-13-0/social-13-0-mail_activity_team/fr/

HviorForgeFlow reviewed (changes requested):
Can you drop empty commits? (the merge ones) Thanks!

atchuthan committed:
[IMP] mail_activity_team: black, isort, prettier

atchuthan committed:
[MIG] mail_activity_team: Migration to 14.0

atchuthan commented:
> Can you drop empty commits? (the merge ones) Thanks!

Removed the commits and pushed the changes once again

pedrobaeza commented:
/ocabot merge nobump

rafaelbn commented:
/ocabot migration mail_activity_team
